pallet/src/lib.rs

```diff
@@ -20,28 +20,59 @@ pub mod pallet {
         sp_std,
     };
     use frame_system::pallet_prelude::*;
-    use pallet_session::Pallet as Session;
+    use pallet_session::{Pallet as Session, SessionManager};
     use primitives::{
         ApiError as AlephApiError, DEFAULT_MILLISECS_PER_BLOCK, DEFAULT_SESSION_PERIOD,
     };
 
+    #[pallet::type_value]
+    pub fn DefaultValidatorsList<T: Config>() -> Option<Vec<T::AccountId>> {
+        None
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn validators_list)]
+    pub type ValidatorsList<T: Config> =
+        StorageValue<_, Option<Vec<T::AccountId>>, ValueQuery, DefaultValidatorsList<T>>;
+
     #[pallet::config]
     pub trait Config: frame_system::Config + pallet_session::Config {
         type AuthorityId: Member
             + Parameter
             + RuntimeAppPublic
             + Default
             + MaybeSerializeDeserialize;
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+    }
+
+    #[pallet::event]
+    #[pallet::metadata(T::AccountId = "AccountId")]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        ChangeValidators(Option<Vec<T::AccountId>>),
     }
 
+    pub struct AlephSessionManager<T>(sp_std::marker::PhantomData<T>);
+
     #[pallet::pallet]
     pub struct Pallet<T>(sp_std::marker::PhantomData<T>);
 
     #[pallet::hooks]
     impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
 
     #[pallet::call]
-    impl<T: Config> Pallet<T> {}
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight(0)]
```

Comment:
Weight should be: `(T::BlockWeights::get().max_block, DispatchClass::Operational)`

Reply:
Done